From 87d68308c7f9f429690e5ccc0e7fd92ccd4e0dd5 Mon Sep 17 00:00:00 2001
From: Om Kate <work.omkate@gmail.com>
Date: Fri, 12 Jun 2026 04:21:25 +0000
Subject: [PATCH 1/3] UPSTREAM: Disable "New Recording" button on non-Wayland
 platforms

Disable (gray out) the "New Recording" button when running on a platform that doesn't support screen recording.

This prevents Spectacle from silently failing on X11 when attempting to screen record.

Show a tooltip on hover explaining to the user that this feature is exclusive to Wayland.

BUG: 509812
---
 src/Gui/RecordingModeMenuButton.qml | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/Gui/RecordingModeMenuButton.qml b/src/Gui/RecordingModeMenuButton.qml
index 54d40666..7692088e 100644
--- a/src/Gui/RecordingModeMenuButton.qml
+++ b/src/Gui/RecordingModeMenuButton.qml
@@ -7,9 +7,13 @@ import QtQuick.Controls as QQC
 import org.kde.spectacle.private
 
 TtToolButton {
+    enabled: SpectacleCore.videoPlatform.supportedRecordingModes !== VideoPlatform.NoRecordingModes
     icon.name: "camera-video"
     text: i18nc("@action select new recording mode", "New Recording")
     down: pressed || RecordingModeMenu.visible
     Accessible.role: Accessible.ButtonMenu
     onPressed: RecordingModeMenu.popup(this)
+    hoverEnabled: true
+    QQC.ToolTip.visible: !enabled && hovered
+    QQC.ToolTip.text: i18nc("@info:tooltip", "This feature is not supported on the current platform")
 }
-- 
2.55.0

